kore/src/Log.hs

```diff
@@ -225,7 +226,16 @@ instance Monad m => MonadLog (LoggerT m) where
         logAction <- ask
         let entryLogger = fromLogAction @ActualEntry logAction
         Monad.Trans.lift $ entryLogger <& toEntry entry
-    logWhile _ = id
+
+    logWhile entry2 action = do
+        logEntry entry2
+        LoggerT . local modifyContext $ getLoggerT action
+      where
+        modifyContext = Colog.cmap $ \entry1 ->
+            entry1
+                { entryContext =
+                    entryContext entry1 <> Seq.singleton (toEntry entry2)
+                }
```

Comment:
You don't have to make this change, but it's the ideal use-case for lenses:

```suggestion
        modifyContext = Colog.cmap $ Lens.(<>~) (field @"entryContext") (Seq.singleton $ toEntry entry2)
```

Reply:
Thanks for the suggestion! For some reason, I wasn't able to get this to compile, but I'll keep it in mind next time I need to implement this functionality.

Reply:
I probably made an embarrassing typo somewhere, I'm sorry! :sweat_smile: